linked_list/implementation.py

```diff
@@ -1,40 +1,127 @@
 from .interface import AbstractLinkedList
-
+from .node import Node
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
 
     def __init__(self, elements=None):
-        pass
+        self.start = None
+        self.end = None
+        if elements:
+            for e in elements:
+                node = Node(e)
+                if not self.start:
+                    self.start = node
+                if self.end:
+                    self.end.next = node
+                self.end = node
 
     def __str__(self):
-        pass
+        str_list = []
```

Comment:
I don't think it was in the spirit of the assignment to use empty brackets `[]`, because the challenge was to implement a list function. So, in theory, those brackets don't exist yet. So it would have been better to call another instance of the class. Something like:
```python
str_list = LinkedList()
```
That being said, maybe I'm wrong.